regolith/file_system.go

```diff
@@ -872,3 +873,110 @@ func MoveOrCopy(
 	}
 	return nil
 }
+
+// SyncDirectories copies the source to destination while checking size and modification time.
+func SyncDirectories(
+	source string, destination string, makeReadOnly bool, copyParentAcl bool,
+) error {
+	// Make destination parent if not exists
+	destinationParent := filepath.Dir(destination)
+	if _, err := os.Stat(destinationParent); os.IsNotExist(err) {
+		err = os.MkdirAll(destinationParent, 0755)
+		if err != nil {
+			return burrito.WrapErrorf(
+				err, osMkdirError, destinationParent)
+		}
+	}
```

Comment:
"If path is already a directory, MkdirAll does nothing and returns nil." - os.Stat is unnecessary.